### PR TITLE
EVAKA-4203: vasu template editor v1

### PR DESCRIPTION
**Pull request by Joosakur**

### CODE REVIEW

File: frontend/src/employee-frontend/components/Header.tsx [hunk 1]
```diff
@@ -263,6 +263,17 @@ const Header = React.memo(function Header({ location }: RouteComponentProps) {
                     </Link>
                   </RequireRole>
                 ) : null}
+                {featureFlags.vasu ? (
```
Comment (kimifoud) [inline frontend/src/employee-frontend/components/Header.tsx:266]: `featureFlags.vasu && <..` would be more idiomatic, but I see that ternary has been used in this file already.

File: frontend/src/employee-frontend/components/vasu/templates/CreateQuestionModal.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+          type: 'MULTISELECT',
+          name: name,
+          optionNames: options,
+          minSelections: 0,
+          maxSelections: null,
+          value: []
+        }
+      default:
+        throw Error('Unexpected type')
+    }
```
Comment (kimifoud) [inline frontend/src/employee-frontend/components/vasu/templates/CreateQuestionModal.tsx:72]: I don't know how our typescript compiler / linter is configured, but I prefer switch without default case and handling all cases explicitly and throwing after the switch. This way the compiler might warn about an unhandled case when a new type is added.

**Review by kimifoud — APPROVED** [review]
Magnificent work! IMO it's good to merge now to avoid further conflicts and smooth the rough edges afterwards.

File: frontend/src/employee-frontend/components/vasu/templates/CreateQuestionModal.tsx (hunk 2 above)
Comment (Joosakur) [reply frontend/src/employee-frontend/components/vasu/templates/CreateQuestionModal.tsx:72]: ah, didn't work without default before but adding `as const` to the enum helped

**Pull request merged**